packages/paima-sdk/paima-mw-core/src/wallet-connect/index.ts

```diff
@@ -0,0 +1,180 @@
+  try {
+    const result = await postConciseData(conciseBuilder.build(), errorFxn);
+    if (!result.success) {
+      return errorFxn(PaimaMiddlewareErrorCode.ERROR_POSTING_TO_CHAIN);
+    }
+    const blockHeightTX = result.blockHeight;
+    await awaitBlock(blockHeightTX);
+    return { success: true, result };
+  } catch (err) {
+    return errorFxn(PaimaMiddlewareErrorCode.ERROR_POSTING_TO_CHAIN, err);
+  }
+}
+
+export async function walletConnectMigrate(
+  from: string,
+  to: string,
+  from_signature: string,
+  to_signature: string
+): Promise<SuccessfulResult<PostDataResponse> | FailedResult> {
+  const errorFxn = buildEndpointErrorFxn('wallet-connect');
+
+  // migrate = &wm|from|to|from_signature|to_signature
+  const conciseBuilder = builder.initialize();
+  conciseBuilder.setPrefix('&wm');
+  conciseBuilder.addValue({ value: from });
+  conciseBuilder.addValue({ value: to });
+  conciseBuilder.addValue({ value: from_signature });
+  conciseBuilder.addValue({ value: to_signature });
+  try {
+    const result = await postConciseData(conciseBuilder.build(), errorFxn);
+    if (!result.success) {
+      return errorFxn(PaimaMiddlewareErrorCode.ERROR_POSTING_TO_CHAIN);
+    }
+    const blockHeightTX = result.blockHeight;
+    await awaitBlock(blockHeightTX);
+    return { success: true, result };
+  } catch (err) {
+    return errorFxn(PaimaMiddlewareErrorCode.ERROR_POSTING_TO_CHAIN, err);
+  }
+}
+
+export async function walletConnectCancelDelegations(
+  to: string,
+  to_signature: string
+): Promise<SuccessfulResult<PostDataResponse> | FailedResult> {
+  const errorFxn = buildEndpointErrorFxn('wallet-cancel-delegations');
+
+  // walletConnect = &wc|to|to_signature
+  const conciseBuilder = builder.initialize();
+  conciseBuilder.setPrefix('&wc');
+  conciseBuilder.addValue({ value: to });
+  conciseBuilder.addValue({ value: to_signature });
+  try {
+    const result = await postConciseData(conciseBuilder.build(), errorFxn);
+    if (!result.success) {
+      return errorFxn(PaimaMiddlewareErrorCode.ERROR_POSTING_TO_CHAIN);
+    }
+    const blockHeightTX = result.blockHeight;
+    await awaitBlock(blockHeightTX);
+    return { success: true, result };
+  } catch (err) {
+    return errorFxn(PaimaMiddlewareErrorCode.ERROR_POSTING_TO_CHAIN, err);
+  }
+}
+
+export class WalletConnectHelper {
+  private static readonly DELEGATE_WALLET_PREFIX = 'DELEGATE-WALLET';
+  private static readonly SEP = ':';
+
+  public buildMessageToSign(subMessage: string): string {
+    return `${WalletConnectHelper.DELEGATE_WALLET_PREFIX}${
+      WalletConnectHelper.SEP
+    }${subMessage.toLocaleLowerCase()}${WalletConnectHelper.SEP}${ENV.CONTRACT_ADDRESS}`;
+  }
+
+  private async signWithExternalWallet(walletType: AddressType, message: string): Promise<string> {
+    switch (walletType) {
+      case AddressType.EVM:
+        return (await EvmInjectedConnector.instance().getProvider()?.signMessage(message)) || '';
```

Comment:
Is returning `''` here really valid? It looks to me like it would cause things to break, so it's better to fail fast than to silently return the wrong result and fail much later imo